pkg/storage/local.go

```diff
@@ -1673,12 +1690,23 @@ func ifOlderThan(imgStore *ImageStoreLocal, repo string, delay time.Duration) ca
 }
 
 func DirExists(d string) bool {
-	fi, err := os.Stat(d)
+	if !utf8.ValidString(d) {
+		return false
+	}
+
+	fileInfo, err := os.Stat(d)
+	if err != nil {
+		if e, ok := err.(*fs.PathError); ok && errors.Is(e.Err, syscall.ENAMETOOLONG) || //nolint: errorlint
+			errors.Is(e.Err, syscall.EINVAL) {
```

Comment:
you can cover this in storage/local_test.go?
by running DirExists() with a long name? :D

Reply:
yes, I created a test with a long, invalid path. now waiting for checks to pass.